src/software/ai/evaluation/ball_test.cpp

```diff
@@ -147,3 +147,49 @@ INSTANTIATE_TEST_CASE_P(Positions, BallPositionsInEnemyCornerOffField,
                         ::testing::Values(Point(4.6, 3), Point(4.6, -3), Point(5.0, 3),
                                           Point(5.0, -3), Point(4.5, -3.1),
                                           Point(4.0, -3.1)));
+TEST(DetectSuccessfulKickTest, ball_over_speed_threshold_and_no_direction_difference)
+{
+    Ball ball({0, 0}, {5, 5}, Timestamp::fromSeconds(0));
+
+    Vector expected_direction(4, 4);
+
+    EXPECT_TRUE(hasBallBeenKicked(ball, expected_direction.orientation()));
+}
+
+TEST(DetectSuccessfulKickTest, ball_under_speed_threshold_and_no_direction_difference)
+{
+    Ball ball({5, 2}, {0, 0.2}, Timestamp::fromSeconds(0));
+
+    Angle expected_direction = Angle::fromDegrees(90);
+
+    EXPECT_FALSE(hasBallBeenKicked(ball, expected_direction));
+}
+
+TEST(DetectSuccessfulKickTest,
+     ball_over_speed_threshold_and_small_negative_direction_difference)
+{
+    Ball ball({0, 1}, {3, 0}, Timestamp::fromSeconds(0));
+
+    Angle expected_direction = Angle::fromDegrees(-15);
+
+    EXPECT_TRUE(hasBallBeenKicked(ball, expected_direction));
+}
+
+TEST(DetectSuccessfulKickTest,
+     ball_over_speed_threshold_and_small_positive_direction_difference)
+{
+    Ball ball({0, 1}, {3, 0}, Timestamp::fromSeconds(0));
+
+    Angle expected_direction = Angle::fromDegrees(15);
+
+    EXPECT_TRUE(hasBallBeenKicked(ball, expected_direction));
+}
+
+TEST(DetectSuccessfulKickTest, ball_over_speed_threshold_and_exceeds_direction_threshold)
+{
+    Ball ball({-5, 2}, {-3, 0}, Timestamp::fromSeconds(0));
+
+    Angle expected_direction = Angle::fromRadians(0);
+
+    EXPECT_FALSE(hasBallBeenKicked(ball, expected_direction));
+}
```

Comment:
can we add a test where the difference in direction is greater than 20?

Reply:
there is a test where the velocity angle is -180 and the `expected_direction` is 0 degrees, but the speed is over the threshold. 

I added a test where the angle difference is greater than 20, but speed is under the threshold.